### DISCUSSION TITLE
Implement a "Fix All" button in "Developer Tools > Statistics" to remove orphaned IDs

### DISCUSSION THREAD

**Discussion by loomyr** [home-assistant/frontend#12644] [Other feature requests]
Could it be possible in a future release to implement a “Fix All” button in the “Developer Tools > Statistics” to clean up all orphaned IDs?

If we face a lot of statistic issues to be fixed (in my case about 7000), and we don’t want to bother by clicking thousands of times the “Fix Issue” button, and clicking the “Remove” button in the dialog then a “Fix All” button could useful.

I was able to clean it manually and wrote a little [how-to](https://community.home-assistant.io/t/statistics-fix-issue-large-amount-of-data/419728) in community guides, however a “Fix All” button might be safer than going to modify it directly within the database.

**Comment by bcutter** [comment]
Would be great to have this. I disabled a lot of entities which produced many thousand rows in the database (including statistics and statistics_short_term tables) which I want to remove now. Note that I disabled them, they are not deleted.

Overview of current situation (scenarios and options) and other related feature requests:

https://community.home-assistant.io/t/remove-statistics-of-disabled-not-deleted-entities-from-recorder/462599

**Comment by teskanoo** [comment]
This is some pretty low hanging fruit as far as I can see - something that with a small amount of work reaps great rewards

**Reply by bcutter** [reply]
...nevertheless unfortunately nobody seems to be willing or able to do that _small amout of work_.

**Comment by bcutter** [comment]
Ping & Push

**Comment by Cryotize** [comment]
I also ran into this Issue with one of my statistics, would be a great feature!

**Comment by gravelfreeman** [comment]
Home Assistant have terrible stuff going on under the hood. This is an example of amateurism. Home automation is to save time but now all the users have to delete 1 by 1 thousands of leftovers. This is terrible.

**Comment by enerbe** [comment]
would be great to have this feature.

**Comment by patrick-foubert-56770** [comment]
+1 in need it !

**Comment by miwa4711** [comment]
+1 love it!

**Comment by turbolooser** [comment]
please....

**Comment by wiedwo** [comment]
Would be a great help.. thanks in advance

**Comment by skybarberom** [comment]
+1

**Comment by mukaschultze** [comment]
https://github.com/home-assistant/frontend/pull/20389

**Comment by gustafssone** [comment]
+1

**Comment by github0123456789** [comment]
+1

**Comment by Zeronozero** [comment]
+1

**Comment by mweedon** [comment]
+1 please

**Comment by jwmcintosh** [comment]
+1

**Comment by xxgmxx** [comment]
+1 I am using autoclicker for an hour now :D

**Comment by kthofmann** [comment]
Same her, would be more than happy if someone could implement a "purge all orphaned statistics" button. 

Thanks in advance.

**Comment by b2un0** [comment]
+1

**Comment by johntdyer** [comment]
Seems this [community guide mentioned in the original post](https://community.home-assistant.io/t/statistics-fix-issue-large-amount-of-data/419728) no longer exists or has been made private

**Reply by bcutter** [reply]
> Seems this [community guide mentioned in the original post](https://community.home-assistant.io/t/statistics-fix-issue-large-amount-of-data/419728) no longer exists or has been made private 

Here's a snapshot: https://web.archive.org/web/20230401222744/https://community.home-assistant.io/t/statistics-fix-issue-large-amount-of-data/419728

**Comment by scobby** [comment]
+1

**Comment by labanex** [comment]
I vote for this as well. I got over 1500 entries that need to be done. FIX ALL would be nice.

**Comment by karwosts** [comment]
In 2024.10 you can mass delete statistics in developer-tools.

**Comment by kwkid** [comment]
regarding @karwosts comment above "In 2024.10 you can mass delete statistics in developer-tools.".
can someone share how this is done?

**Reply by karwosts** [reply]
![image](https://github.com/user-attachments/assets/7764f4c5-6f59-427b-ad0e-1a9f23d650dd)
![image](https://github.com/user-attachments/assets/93780592-0bbb-47f8-b564-f2f8a7ce7fcb)

**Reply by bcutter** [reply]
Yeah well, as mentioned, that's a "delete all" instead of "fix all" button. Deletion is not always the desired action. Unity conversions etc. happen a lot.
